test/integration/protocol_integration_test.cc

```diff
@@ -5096,12 +5096,12 @@ TEST_P(ProtocolIntegrationTest, ServerHalfCloseBeforeClientWithErrorAndBufferedR
     } else if (downstreamProtocol() == Http::CodecType::HTTP2) {
       ASSERT_TRUE(response->waitForReset());
     } else if (downstreamProtocol() == Http::CodecType::HTTP3) {
-      // Unlike H/2 codec H/3 codec attempts to send pending data before the reset
-      // So it needs window to push the data to the client and then the reset
-      // which just gets discarded since end_stream has been received just before
-      // reset.
+      // Unlike H/2, H/3 client codec only stops sending request but still attempts to finish
+      // receiving response, so resume reading.
```

Comment:
I'm not quite sure that I understand this comment. In particular, I'm not sure I understand what "resume reading" means in this case. Is this basically saying that the H3 code will (potentially) read the entire response and that's why we need to call "waitForEndStream()"? But then eventually the server will also send STOP_SENDING, which we *also* need to wait for? I'm slightly surprised that we need to wait for both, and not just wait for the reset. Do you understand why we need to wait for both?

Reply:
We don't have to wait for the response, but we definitely need to wait for the reset.

Reply:
Actually we need to wait for both because receiving STOP_SENDING without finishing read the response won't close the QUIC stream. That's what made the test flaky.

Reply:
Ah, ok. Thanks, that makes sense. In that case, shall we update the comment to explain that in this case the H3 codec needs to wait for both the end of the response and the STOP_SENDING in order for the stream to be fully closed?

Reply:
done